plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -878,23 +878,26 @@ func haveSame(a1, a2 []string) bool {
 
 func GeneralPredicates(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
 	var predicateFails []algorithm.PredicateFailureReason
-	fit, reasons, err := noncriticalPredicates(pod, meta, nodeInfo)
+	fit, reasons, err := EssentialPredicates(pod, meta, nodeInfo)
 	if err != nil {
 		return false, predicateFails, err
 	}
 	if !fit {
+		// No need check other predicates if the essential predicates failed
 		predicateFails = append(predicateFails, reasons...)
+		return len(predicateFails) == 0, predicateFails, nil
```

Comment:
We are explicitly computing all predicated and avoid early return to give a more actionable feedback for user.
Basically, if you return only first failed predicate, then user may expect that once he fixes this issue it will schedule. But it may not be the case.
If we return e.g. 6 different predicates failing on each node, user expects that it won't be easy to schedule this pod and will need to take a bunch of actions.